### PR TITLE
dump reload commands v0.2

### PR DESCRIPTION
**Pull request by snopoke**
This PR includes:
- tools for re-sharding data (getting the shard for a doc_id)
- dump data command to dump a domains data to disk (based off [dumpdata](https://docs.djangoproject.com/en/1.10/ref/django-admin/#dumpdata))
  - currently only for a few SQL models
- load domains data from disk into the db (based off [loaddata](https://docs.djangoproject.com/en/1.10/ref/django-admin/#loaddata))
  - currently only for a few SQL models

@biyeun 
@dimagi/scale-team 

revised version of https://github.com/dimagi/commcare-hq/pull/13734

### CODE REVIEW

File: corehq/apps/dump_reload/tests/test_sql_dump_load.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        post_rule = AutomaticUpdateRule.objects.get(pk=pre_rule.pk)
+        post_criteria = AutomaticUpdateRuleCriteria.objects.get(pk=pre_criteria.pk)
+        post_action_update = AutomaticUpdateAction.objects.get(pk=pre_action_update.pk)
+        post_action_close = AutomaticUpdateAction.objects.get(pk=pre_action_close.pk)
+
+        self.assertModelsEqual(
+            [pre_rule, pre_criteria, pre_action_update, pre_action_close],
+            [post_rule, post_criteria, post_action_update, post_action_close]
+        )
+
```
Comment (dimagimon) [inline corehq/apps/dump_reload/tests/test_sql_dump_load.py]: W391 blank line at end of file

File: corehq/form_processor/tests/test_sharding.py [hunk 2]
```diff
@@ -72,3 +73,102 @@ def test_objects_distributed_to_all_dbs(self):
...
+    },
+    'groups': {
+        'main': ['default'],
+        'proxy': ['proxy'],
+        'form_processing': ['p1', 'p2', 'p3', 'p4', 'p5'],
+    }
+}
+
+
+@override_settings(PARTITION_DATABASE_CONFIG=PARTITION_DATABASE_CONFIG, DATABASES=DATABASES, ALLOW_FORM_PROCESSING_QUERIES=True)
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_sharding.py]: E501 line too long (128 > 115 characters)

File: corehq/apps/dump_reload/sql/dump.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
+from collections import OrderedDict
+
+from django.apps import apps
+from django.conf import settings
+from django.core import serializers
+from django.db import router
+from django.db.models import Q
```
Comment (dimagimon) [inline corehq/apps/dump_reload/sql/dump.py]: F401 'Q' imported but unused

File: corehq/apps/users/signals.py [hunk 4]
```diff
@@ -24,8 +24,10 @@ def set_language(sender, **kwargs):
 
 
 # Signal that syncs django_user => couch_user
-def django_user_post_save_signal(sender, instance, created, **kwargs):
+def django_user_post_save_signal(sender, instance, created, raw=False, **kwargs):
     from corehq.apps.users.models import CouchUser
+    if raw:
```
Comment (snopoke) [inline corehq/apps/users/signals.py]: When re-loading dumped data into the DB the post_save signal is called with raw=True.

**Comment by snopoke** [comment]
@emord :white_check_mark: finally

**Pull request merged**